src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
+        }
+    }
+
+    public void checkIndexWithSize(int index) {
+        if (index == size) {
+            throw new ArrayIndexOutOfBoundsException(IOOB_MESSAGE);
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.setNext(newNode);
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (size == index) {
+            add(value);
+        } else {
+            Node<T> newNode = new Node<>(null, value, null);
+            Node<T> currentNode = getCurrentNode(index);
+            if (index == 0) {
+                newNode.setNext(currentNode);
+                currentNode.setPrev(newNode);
+                first = newNode;
+            } else {
+                currentNode.getPrev().setNext(newNode);
+                newNode.setPrev(currentNode.getPrev());
+                newNode.setNext(currentNode);
+                currentNode.setPrev(newNode);
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value: list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexWithSize(index);
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        return currentNode.getValue();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        checkIndexWithSize(index);
+        Node<T> currentNode = getCurrentNode(index);
+        T oldValue = currentNode.getValue();
+        currentNode.setValue(value);
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexWithSize(index);
+        checkIndex(index);
+        Node<T> currentNode = getCurrentNode(index);
+        if (index == 0 && size == 1) {
+            first = null;
+            last = null;
+        } else {
+            if (index == 0) {
+                currentNode.getNext().setPrev(null);
+                first = currentNode.getNext();
+            } else if (index == size - 1) {
+                currentNode.getPrev().setNext(null);
+                last = currentNode.getPrev();
+            } else {
+                currentNode.getPrev().setNext(currentNode.getNext());
+                currentNode.getNext().setPrev(currentNode.getPrev());
+            }
+        }
+        size--;
+        return currentNode.getValue();
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int currentIndex = getCurrentIndex(object);
+        if (currentIndex == -1) {
+            return false;
+        }
+        remove(currentIndex);
+        return true;
```

Comment:
lets iterate over the list here, and use unlink method to remove the node, because you are iterating 2 times
in getCurrentIndex an remove

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/81781479/207065304-856bc7cf-d32e-4a79-9b13-c3fb252ca21d.png">